O/OpenBLAS/OpenBLAS@0.3.9/bundled/patches/openblas-winexit.patch

```diff
@@ -0,0 +1,332 @@
+From f919c3301fabbaa5d965dcc7b1c3d6892a8c730a Mon Sep 17 00:00:00 2001
+From: Keno Fischer <keno@juliacomputing.com>
+Date: Sat, 14 Mar 2020 12:05:19 +0100
+
+---
+ driver/others/memory.c | 171 ++++++-----------------------------------
+ 1 file changed, 22 insertions(+), 149 deletions(-)
+
+diff --git a/driver/others/memory.c b/driver/others/memory.c
+index 62a5a021..1eec2c0b 100644
+--- a/driver/others/memory.c
++++ b/driver/others/memory.c
+@@ -80,7 +80,7 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ #undef COMPILE_TLS
+ #endif
+ 
+-#if defined(__GLIBC_PREREQ) 
++#if defined(__GLIBC_PREREQ)
+ #if !__GLIBC_PREREQ(2,20)
+ #undef COMPILE_TLS
+ #endif
+@@ -248,7 +248,7 @@ int get_num_procs(void) {
+       return nums;
+     }
+     ret = CPU_COUNT_S(size,cpusetp);
+-    if (ret > 0 && ret < nums) nums = ret;	
++    if (ret > 0 && ret < nums) nums = ret;
+     CPU_FREE(cpusetp);
+     return nums;
+   } else {
+@@ -257,7 +257,7 @@ int get_num_procs(void) {
+       return nums;
+     }
+     ret = CPU_COUNT(&cpuset);
+-    if (ret > 0 && ret < nums) nums = ret;	
++    if (ret > 0 && ret < nums) nums = ret;
+     return nums;
+   }
+  #endif
+@@ -1510,7 +1510,7 @@ void CONSTRUCTOR gotoblas_init(void) {
+ 
+ }
+ 
+-void DESTRUCTOR gotoblas_quit(void) {
++void gotoblas_quit(void) {
+ 
+   if (gotoblas_initialized == 0) return;
+ 
+@@ -1521,7 +1521,7 @@ void DESTRUCTOR gotoblas_quit(void) {
+   TlsFree(local_storage_key);
+ #else
+   pthread_key_delete(local_storage_key);
+-#endif		
++#endif
+ #endif
+ 
+ #ifdef PROFILE
+@@ -1547,74 +1547,12 @@ void DESTRUCTOR gotoblas_quit(void) {
+ #endif
+ }
+ 
+-#if defined(_MSC_VER) && !defined(__clang__)
```

Comment:
@Keno I think your patch was included in OpenBLAS 0.3.9 already (https://github.com/xianyi/OpenBLAS/commit/23f322f997c8b018977be24122c56fb62d728a05), except for those lines. Otherwise it seems to be just formatting changes. Should I reduce it just to removing those lines / is this still relevant?

Reply:
@keno Could you review? Would be nice to get this merged.

Reply:
Yes, most of this patch still applies. You can remove the part that was already applied, but other than that it's still needed. Somebody needs to work with upstream to get this all cleaned up, but I haven't had the time so far.

Reply:
Thanks! @Keno, checked out 0.3.7, applied the patch, rebased to 0.3.9, and extracted the patch again, so should be fine.

The new commit below basically just removes all the autoformatting / whitespace changes from the patch that are not really important I imagine.